tests/versioning.go

```diff
@@ -3591,8 +3591,14 @@ func (s *VersioningIntegSuite) dispatchCron(newVersioning bool) {
 	}, "wf")
 	s.NoError(err)
 
-	// give it ~3 runs on v1
-	time.Sleep(3500 * time.Millisecond)
+	// give it >=3 runs on v1
+	s.Eventually(
+		func() bool {
+			return runs1.Load() >= int32(3)
+		},
+		3500*time.Millisecond,
```

Comment:
The Eventually will keep evaluating vs the old code would wait 3.5s then check. 
I think using Eventually would not make this more likely to pass, unless you allow longer duration.

Reply:
Replacing the sleep with eventually is just an optimization to make the test potentially pass quicker, and to make the test say what it's waiting for, thereby making it more readable than a sleep.

The code change that fixes the test flake is making sure that w2 is closed and stops appending to `runIds2` before we read from `runIds2` in the for-loop at the end. I should have made that more clear in the description